### DISCUSSION TITLE
Add custom prefix option for `/tf` to the Studio to enable URDF to follow `/tf` messages from `.bag` file

### DISCUSSION THREAD

**Discussion by akruzliak** [foxglove/community#759] [Q&A] [answered]
# Current state
Having recorded a `.bag` file with rosbag in ROS1, I am unable to move a loaded URDF model with the `/tf` messages unless I manually change the names of joints in the mentioned URDF file, e.g., using `xacro`. The same goes for having the `/robot_description` topic (automatically published by `robot_state_publisher` in ROS2, I emulated this behavior by writing a publisher that publishes the robot URDF in XML format as a string to the `/robot_description` topic that is then recorded in the `.bag` file.

# Feature request
Add an option to the 3D viewer's options to choose a prefix (namespace and/or tf_prefix) for the `/tf` that would be either automatically removed from the `/tf` messages or added to the URDF model provided as a Custom Layer in the 3D viewer options or as a `/robot_description` topic, if it exists in the `.bag`/`mcap` file. Currently, it works like this in RVIZ and would enable GREAT support for people wanting to omit problematic `rqt_bag` viewer in favor of Foxglove Studio. 

RVIZ option: 
![image](https://github.com/foxglove/community/assets/64916531/3a6fcde0-ca6a-4e0d-afca-784aa64bcfb7)

I believe this is an important option for robotics visualizations, as almost everyone in mobile robotics, maybe with the exception of autonomous driving, uses URDF models.

I open a similar discussion to [#407](https://github.com/orgs/foxglove/discussions/407#discussion-4668932) because they only talk about namespaces and ROS2 bridge; I talk about namespaces AND tf_prefix, using ROS1 and a `.bag` file. And also I think it is an important visualization feature, so I would like to bring more spotlight.

I also link Slack discussion detailing this problem: [Slack link](https://foxglove.slack.com/archives/C02RFDCG5NK/p1687768396126929)

**Comment by achim-k** [comment, marked as answer]
The frame prefix setting has been added in https://github.com/foxglove/studio/pull/6359 and is available as of Studio v1.62.0
